Bug: Extracting Model Information for tokenizer.chat_template

When using this model:

https://huggingface.co/Qwen/Qwen3-8B-GGUF?show_file_info=Qwen3-8B-Q8_0.gguf

When making this call:

```
llama.ModelMetaValStr(mdl, "tokenizer.chat_template")
```

I am getting this panic:

```
panic: runtime error: slice bounds out of range [:17245] with capacity 8192
```

It is coming from this function and specificially:

```
	// copy to a new slice to avoid retaining the entire buffer
	value := make([]byte, int32(result))
	copy(value, buf[:int32(result)])
```

```
func ModelMetaValStr(model Model, key string) (string, bool) {
	if model == 0 {
		return "", false
	}
	buf := make([]byte, 8192)
	b := unsafe.SliceData(buf)
	bLen := int32(len(buf))

	keyPtr, _ := utils.BytePtrFromString(key)
	var result ffi.Arg
	modelMetaValStrFunc.Call(
		unsafe.Pointer(&result),
		unsafe.Pointer(&model),
		unsafe.Pointer(&keyPtr),
		unsafe.Pointer(&b),
		&bLen,
	)
	if int32(result) < 0 {
		return "", false
	}

	// copy to a new slice to avoid retaining the entire buffer
	value := make([]byte, int32(result))
	copy(value, buf[:int32(result)])

	return string(value), true
}
```

And it makes no sense since this is the template:

```
{%- if tools %} {{- '<|im_start|>system\n' }} {%- if messages[0].role == 'system' %} {{- messages[0].content + '\n\n' }} {%- endif %} {{- "# 
Tools\n\nYou may call one or more functions to assist with the user query.\n\nYou are provided with function signatures within <tools></tools> 
XML tags:\n<tools>" }} {%- for tool in tools %} {{- "\n" }} {{- tool | tojson }} {%- endfor %} {{- "\n</tools>\n\nFor each function call, return a json 
object with function name and arguments within <tool_call></tool_call> XML tags:\n<tool_call>\n{\"name\": <function-name>, \"arguments\": 
<args-json-object>}\n</tool_call><|im_end|>\n" }} {%- else %} {%- if messages[0].role == 'system' %} {{- '<|im_start|>system\n' + 
messages[0].content + '<|im_end|>\n' }} {%- endif %} {%- endif %} {%- set ns = namespace(multi_step_tool=true, 
last_query_index=messages|length - 1) %} {%- for index in range(ns.last_query_index, -1, -1) %} {%- set message = messages[index] %} {%- if 
ns.multi_step_tool and message.role == "user" and not('<tool_response>' in message.content and '</tool_response>' in message.content) %} 
{%- set ns.multi_step_tool = false %} {%- set ns.last_query_index = index %} {%- endif %} {%- endfor %} {%- for message in messages %} {%- 
if (message.role == "user") or (message.role == "system" and not loop.first) %} {{- '<|im_start|>' + message.role + '\n' + message.content + 
'<|im_end|>' + '\n' }} {%- elif message.role == "assistant" %} {%- set content = message.content %} {%- set reasoning_content = '' %} {%- if 
message.reasoning_content is defined and message.reasoning_content is not none %} {%- set reasoning_content = 
message.reasoning_content %} {%- else %} {%- if '</think>' in message.content %} {%- set content = message.content.split('</think>')
[-1].lstrip('\n') %} {%- set reasoning_content = message.content.split('</think>')[0].rstrip('\n').split('<think>')[-1].lstrip('\n') %} {%- endif %} 
{%- endif %} {%- if loop.index0 > ns.last_query_index %} {%- if loop.last or (not loop.last and reasoning_content) %} {{- '<|im_start|>' + 
message.role + '\n<think>\n' + reasoning_content.strip('\n') + '\n</think>\n\n' + content.lstrip('\n') }} {%- else %} {{- '<|im_start|>' + 
message.role + '\n' + content }} {%- endif %} {%- else %} {{- '<|im_start|>' + message.role + '\n' + content }} {%- endif %} {%- if 
message.tool_calls %} {%- for tool_call in message.tool_calls %} {%- if (loop.first and content) or (not loop.first) %} {{- '\n' }} {%- endif %} {%- 
if tool_call.function %} {%- set tool_call = tool_call.function %} {%- endif %} {{- '<tool_call>\n{"name": "' }} {{- tool_call.name }} {{- '", 
"arguments": ' }} {%- if tool_call.arguments is string %} {{- tool_call.arguments }} {%- else %} {{- tool_call.arguments | tojson }} {%- endif %} {{- 
'}\n</tool_call>' }} {%- endfor %} {%- endif %} {{- '<|im_end|>\n' }} {%- elif message.role == "tool" %} {%- if loop.first or (messages[loop.index0 
- 1].role != "tool") %} {{- '<|im_start|>user' }} {%- endif %} {{- '\n<tool_response>\n' }} {{- message.content }} {{- '\n</tool_response>' }} {%- if 
- loop.last or (messages[loop.index0 + 1].role != "tool") %} {{- '<|im_end|>\n' }} {%- endif %} {%- endif %} {%- endfor %} {%- if 
- add_generation_prompt %} {{- '<|im_start|>assistant\n' }} {%- if enable_thinking is defined and enable_thinking is false %} {{- 
- '<think>\n\n</think>\n\n' }} {%- endif %} {%- endif %}
```

--- a/pkg/llama/lora.go
+++ b/pkg/llama/lora.go
@@ -145,7 +145,7 @@ func AdapterMetaValStr(adapter AdapterLora, key string) (string, bool) {
 	if adapter == 0 {
 		return "", false
 	}
-	buf := make([]byte, 8192)
+	buf := make([]byte, 32768)
 	b := unsafe.SliceData(buf)
 	bLen := int32(len(buf))
 
@@ -211,7 +211,7 @@ func AdapterMetaValStrByIndex(adapter AdapterLora, i int32) (string, bool) {
 	if adapter == 0 {
 		return "", false
 	}
-	buf := make([]byte, 8192)
+	buf := make([]byte, 32768)
 	b := unsafe.SliceData(buf)
 	bLen := int32(len(buf))
 

--- a/pkg/llama/model.go
+++ b/pkg/llama/model.go
@@ -586,7 +586,7 @@ func ModelMetaValStr(model Model, key string) (string, bool) {
 	if model == 0 {
 		return "", false
 	}
-	buf := make([]byte, 8192)
+	buf := make([]byte, 32768)
 	b := unsafe.SliceData(buf)
 	bLen := int32(len(buf))
 
@@ -654,7 +654,7 @@ func ModelMetaValStrByIndex(model Model, i int32) (string, bool) {
 	if model == 0 {
 		return "", false
 	}
-	buf := make([]byte, 8192)
+	buf := make([]byte, 32768)
 	b := unsafe.SliceData(buf)
 	bLen := int32(len(buf))
 

